bundle/src/main/java/com/adobe/acs/commons/granite/ui/components/include/IncludeDecoratorFilterImpl.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2017 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.granite.ui.components.include;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.engine.EngineConstants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import javax.annotation.CheckForNull;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+
+
+import java.io.IOException;
+
+@Component(
+        service = Filter.class,
+        configurationPolicy = ConfigurationPolicy.OPTIONAL,
```

Comment:
Should this be Optional? Wont this add this Filter to the Request processing chain even for people that arent using this feature?

Reply:
@justinedelson WDYT? while this Filter would become active w/out someone using it, it does scope to resource type so it shouldn't be invoked unless someone intentionally apply the resource type...  Don't want to be too dogmatic; feels like a grey area wrt to "disabled by default" - i could go either way.

Reply:
@justinedelson any strong opinion one way or another? if not, i'm inclined to let sleeping dogs lie.

Reply:
I don't fully grok why this is optional. It seems like you are effectively opting in to using this by virtue of referencing the component resource type.

@davidjgonzalez is your concern that a consumer of ACS AEM Commons will use the resource type `acs-commons/granite/ui/components/include` for some _other_ purpose and be surprised to see this filter invoked?

Reply:
@justinedelson my concern (IIUC) is more this Sling Filter will be "registered" into the Sling Filter-stack even if you aren't actively using it (using it, meaning you don't have a resource w/ this resource type). Admittedly it won't (shouldn't) ever be executed unless this resource type is present, but, technically, the code *is* "active", and Sling has to make a decision in request processing to invoke it or not.

Optional vs Required shouldn't change any behaviors, its a more pedantic question; do we want OSGi Components to be active, even if they aren't invoked by Sling, by default - or do we want users to always, explicitly turn things on.

Reply:
Yeah, I don't know what the performance impact is of just having the filter there without using the component. I can see that leaving this as optional is safer. But... what is the behavior when you _use_ the `acs-commons/granite/ui/components/include` resource type and this filter *isn't* active? A wise man once told me that "documentation is a polyfill for confusing design" :)

Reply:
Wise indeed! Let's keep it as is! :P